src/game/g_main.c

```diff
@@ -4298,6 +4322,23 @@ void CheckExitRules(void)
 					// score is tied, so don't end the game
 					return;
 				}
+
+				if (level.suddenDeath)
+				{
+					if (DynamiteOnObjective())
+					{
+						return;
+					}
+					level.suddenDeath = 0;
```

Comment:
Not completely sure about this line, in regard to the change in `g_script_actions.c`.